AK/Random.cpp

```diff
@@ -28,4 +30,16 @@ u32 get_random_uniform(u32 max_bounds)
     return random_value % max_bounds;
 }
 
+u64 get_random_uniform_64(u64 max_bounds)
```

Comment:
What about having a template function for that?

Comments are left as an exercise for the reader :eyes:
```cpp
template<Unsigned T>
requires(!SameAs<T, u256>) // I didn't think too much if it there is an algorithm compatible with u256
T get_random_uniform(T max_bounds)
{
    // I used a u256 as the overkill value, you could create a Promote template to do a finner promotion
    auto const max_usable = NumericLimits<T>::max() - (static_cast<u256>(NumericLimits<T>::max()) + 1) % max_bounds;
    auto random_value = get_random<T>();
    for (int i = 0; i < 20 && random_value > max_usable; ++i) {
        random_value = get_random<T>();
    }
    return random_value % max_bounds;
}
```

Reply:
I'm unsure how future-resistant should this be, nor of any performance implications. There are numbers going into u4096! For the PR I only need the 64bit extension.

Maybe I could use `IsOneOf<u32,u64>` to DRY this instead, and use u128 for the overkill value? IIRC u128 has the chance of being done natively on 64bit CPUs with the right instructions/registers.

Reply:
It's not like this code is in any hot-path so let's not think too much about performance.
I'm perfectly fine with using restricting input to `u32` and `u64`.  You can even use `OneOf` instead of `Unsigned` and remove the `require` clause.

Reply:
If you really want, you could just create 

```
template<Unsigned T>
T get_random_uniform(T max_bounds)
{
    Conditional< .... > max_usable;
    if constexpr (IsSame<T, u64>) {
         max_usable = ....
    }
    else if constexpr (IsSame<T, ...>) {
         max_usable = ....
    }
    auto random_value = get_random<T>();
    for (int i = 0; i < 20 && random_value > max_usable; ++i) {
        random_value = get_random<T>();
    }
    return random_value % max_bounds;
}
```

With some horrible `Conditional<T, ..., Conditional<T, .... Conditional<T, ...>>>` ternary expression to select the type for the max value.

or a simple helper struct to select it

```
template <Unsigned T>
struct MaxBoundTypeFor {
   using Type = u64;
};

template<>
struct MaxBoundTypeFor<u128> {
   using Type = u256;
};

...
template <typename T>
void foo() {
    MaxBoundTypeFor<T> max = NumericLimits<MaxBoundTypeFor<T>>::max()  - 1234;
}
```